src/main/java/org/openstreetmap/atlas/checks/database/FlagDatabaseSubCommand.java

```diff
@@ -0,0 +1,375 @@
+     *         enum value; otherwise, an empty optional.
+     */
+    private Optional<OutputFileType> getOptionalOutputType(final File logFile)
+    {
+        // Note that technically the true extension is just .gz, so we can't use the same method as
+        // below.
+        if (logFile.getName().endsWith(ZIPPED_LOG_EXTENSION))
+        {
+            return Optional.of(OutputFileType.COMPRESSED_LOG);
+        }
+        else if (FilenameUtils.getExtension(logFile.getName()).equals(LOG_EXTENSION))
+        {
+            return Optional.of(OutputFileType.LOG);
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Returns the OSM identifier for a given JsonObject. Atlas Checks OSM identifier changed from
+     * "osmid" to "osmIdentifier"
+     * {@link <a href="https://github.com/osmlab/atlas-checks/pull/116/files">here</a>}
+     *
+     * @param properties
+     *            CheckFlag properties
+     * @return OSM identifier
+     */
+    private long getOsmIdentifier(final JsonObject properties)
+    {
+        return properties.get("osmid") == null ? properties.get("osmIdentifier").getAsLong()
+                : properties.get("osm_id").getAsLong();
+    }
+
+    /**
+     * Read a file that we know we should be able to handle
+     *
+     * @param inputFile
+     *            Some file with a valid, appropriate extension.
+     * @param fileType
+     *            The type of file that inputFile is
+     * @return a BufferedReader to read inputFile
+     * @throws IOException
+     *             if the file is not found or is poorly formatted, given its extension. For
+     *             example, if this file is gzipped and something goes wrong in the unzipping
+     *             process, it might throw an error
+     */
+    private BufferedReader getReader(final File inputFile, final OutputFileType fileType)
+            throws IOException
+    {
+        if (fileType == OutputFileType.LOG)
+        {
+            return new BufferedReader(new FileReader(inputFile.getPath()));
+        }
+        return new BufferedReader(new InputStreamReader(
+                new GZIPInputStream(new FileInputStream(inputFile.getPath()))));
+    }
+
+    /**
+     * Filters non OSM tag in CheckFlag properties and converts into Map object for PostgreSQL
+     * hstore
+     *
+     * @param properties
+     *            CheckFlag properties
+     * @return hstore string
+     */
+    private Map<String, String> getTags(final JsonObject properties)
+    {
+        final Map<String, String> hstore = new HashMap<>();
+        final Set<String> blacklistKeys = new HashSet<>();
+        blacklistKeys.add("itemType");
+        blacklistKeys.add("identifier");
+        blacklistKeys.add("osmid");
+        blacklistKeys.add("osmIdentifier");
+        blacklistKeys.add("relations");
+        blacklistKeys.add("members");
+        blacklistKeys.add(ISO_COUNTRY_CODE);
```

Comment:
We could probably initialize `blacklistKeys` outside of this function so we only have to do it once